packages/core/src/browser/tree/tree-widget.ts

```diff
@@ -170,8 +176,12 @@ export class TreeWidget extends VirtualWidget implements StatefulWidget {
 
     protected renderCaption(node: ITreeNode, props: NodeProps): h.Child {
         const tooltip = this.getDecorationData(node, 'tooltip').filter(notEmpty).join(' • ');
+        const classes = [TREE_NODE_SEGMENT_CLASS];
+        if (!this.hasTrailingSuffixes(node)) {
+            classes.push(TREE_NODE_SEGMENT_GROW_CLASS);
+        }
         let attrs = this.decorateCaption(node, {
-            className: TREE_NODE_CAPTION_CLASS
+            className: this.merge(...classes)
```

Comment:
minor:
won't be `classes.join(' ')` shorter than `this.merge(...classes)`? I think no one will override `this.merge`